proposals/lip-0053.md

```diff
@@ -346,6 +380,119 @@ def updatePartnerChainOutboxRoot(ccu: CCU) -> None:
     channel(ccu.params.sendingChainID).partnerChainOutboxRoot = calculateRootFromRightWitness(inbox.size, inbox.appendPath, messageWitnessHashes)
 ```
 
+#### verifyRoutingRules
+
+```python
```

Comment:
Suggested modification:

```
def verifyRoutingRules(ccu: CCU, ccm: CCM) -> None:
   # Sending and receiving chains must differ.
   if ccm.receivingChainID == ccm.sendingChainID:
       raise Exception("Sending and receiving chains must differ.")

    # Processing via direct channels.
    if chainAccount(ccm.sendingChainID) exists:
        # The CCM must come from the sending chain.
        if ccu.params.sendingChainID != ccm.sendingChainID:
            raise Exception("CCM is not from the sending chain.")
        if ccm.status == CCM_STATUS_CODE_CHANNEL_UNAVAILABLE:
            raise Exception("CCM status channel unavailable can only be set on the mainchain.")

    # Processing on a sidechain.
    if ownChainAccount.chainID != getMainchainID():
        # The CCM must be directed to the sidechain.
        if ownChainAccount.chainID != ccm.receivingChainID:
            raise Exception("CCM is not directed to the sidechain.")
```

Reply:
We prefer to defer this change to the Direct sidechain channels LIP, as it would require minimal effort to implement and sidechains would need to hardfork Interoperability anyway to use direct channels.